source/dub/dub.d

```diff
@@ -730,9 +730,9 @@ class Dub {
 	}
 
 	/// Fetches the package matching the dependency and places it in the specified location.
-	Package fetch(string packageId, const Dependency dep, PlacementLocation location, FetchOptions options, string reason = "")
+	Package fetch(PackageName package_id, const Dependency dep, PlacementLocation location, FetchOptions options, string reason = "")
```

Comment:
I think most changes in this file are breaking the public interface, which means they break reggae.

Reply:
Shall I just revert all the parameters in this package from `PackageName` to `string`?

Reply:
Resolved at d2c30ec9.

Reply:
`private` is okay. For `public` function, you can just change them, but add a wrapper function that is deprecated.
In this example:
```D
deprecated("Use the overload that accepts a `PackageName` instead of a string as first argument")
Package fetch(PackageName id, const Dependency dep, PlacementLocation location, FetchOptions options, string reason = "") {
    return this.fetch(PackageName(id), dep, location, options, reason);
}

/// Ditto
Package fetch(PackageName id, const Dependency dep, PlacementLocation location, FetchOptions options, string reason = "") {
    // Real code
}
```

Reply:
Ok, I'm gonna use the pattern:

```	/// Compatibility overload. Use the version without a `force_remove` argument instead.
	void remove(PackageName name, PlacementLocation location, bool force_remove,
				scope size_t delegate(in Package[] packages) resolve_version)
	{
		remove(name, location, resolve_version);
	}
    /// ditto
	deprecated("Use the overload taking a `PackageName` as first parameter instead ")
	void remove(string name, PlacementLocation location, bool force_remove,
				scope size_t delegate(in Package[] packages) resolve_version)
    {
        remove(PackageName(name), location, force_remove, resolve_version);
    }
```

Reply:
Resolved via 6468bb2f